### PR TITLE
Fixes for related objects & Directory sync

### PR DESCRIPTION
**Pull request by rjzondervan**

### CODE REVIEW

**Review by bbrands02 — APPROVED** [review]

**Review by RalkeyOfficial — APPROVED** [review]

**Review by remko48 — APPROVED** [review]

File: lib/Controller/OrganizationsController.php [hunk 1]
```diff
@@ -109,6 +111,17 @@ public function create(): JSONResponse
         // Save the new organization object
         $object = $this->objectService->saveObject('organization', $data);
 
+        // If object is a class change it to array
+        if (is_object($object)) {
```
Comment (WilcoLouwerse) [inline lib/Controller/OrganizationsController.php:115]: === true

File: lib/Controller/PublicationsController.php [hunk 2]
```diff
@@ -197,6 +201,17 @@ public function create(ObjectService $objectService): JSONResponse
         // Save the new publication object
         $object = $this->objectService->saveObject('publication', $data);
 
+        // If object is a class change it to array
+        if (is_object($object)) {
```
Comment (WilcoLouwerse) [inline lib/Controller/PublicationsController.php:205]: === true

File: lib/Controller/ThemesController.php [hunk 3]
```diff
@@ -11,7 +11,7 @@
 use OCP\AppFramework\Http\JSONResponse;
 use OCP\IAppConfig;
 use OCP\IRequest;
```
Comment (WilcoLouwerse) [inline lib/Controller/ThemesController.php:13]: Does this file have class docblock?

File: lib/Controller/ThemesController.php [hunk 4]
```diff
@@ -97,6 +99,17 @@ public function create(): JSONResponse
         // Save the new theme object
         $object = $this->objectService->saveObject('theme', $data);
 
+        // If object is a class change it to array
+        if (is_object($object)) {
```
Comment (WilcoLouwerse) [inline lib/Controller/ThemesController.php:103]: === true

**Review by WilcoLouwerse — CHANGES_REQUESTED** [review]

File: lib/Service/DirectoryService.php [hunk 5]
```diff
@@ -93,7 +93,7 @@ private function getDirectoryFromListing(Listing|array $listing): array
 //		$listing['id'] = $listing['uuid'];
```
Comment (WilcoLouwerse) [inline lib/Service/DirectoryService.php:93]: Does this service have a class docblock?

**Pull request merged**